selfservice/hook/web_hook.go

```diff
@@ -499,3 +501,18 @@ func isTimeoutError(err error) bool {
 	var te interface{ Timeout() bool }
 	return errors.As(err, &te) && te.Timeout() || errors.Is(err, context.DeadlineExceeded)
 }
+
+func instrumentHTTPClientForEvents(ctx context.Context, httpClient *retryablehttp.Client) {
+	var (
+		attempt = 0
+		reqBody []byte
+	)
+	httpClient.RequestLogHook = func(_ retryablehttp.Logger, req *http.Request, retryNumber int) {
+		attempt = retryNumber + 1
+		reqBody, _ = httputil.DumpRequestOut(req, true)
+	}
+	httpClient.ResponseLogHook = func(_ retryablehttp.Logger, res *http.Response) {
+		resBody, _ := httputil.DumpResponse(res, true)
+		trace.SpanFromContext(ctx).AddEvent(events.NewWebhookDelivered(ctx, res.Request.URL, reqBody, res.StatusCode, resBody, attempt))
```

Comment:
Can we add some kind of unique identifier per webhook dispatch? E.g. a login flow triggers a webhook, which is retried a few times and then succeeds. This leads to multiple events in the storage. However, it's hard to tell which events belong together.

Reply:
Also a good idea!

Reply:
done